tmt/steps/execute/internal.py

```diff
@@ -729,18 +725,20 @@ def _run_tests(
                         if invocation.handle_restart():
                             continue
 
-                    except tmt.utils.RebootTimeoutError:
+                    except tmt.utils.RebootTimeoutError as error:
+                        invocation.exception = error
                         for result in invocation.results:
                             result.result = ResultOutcome.ERROR
-                            result.note.append('reboot timeout')
 
-                    else:
+                    except tmt.utils.ReconnectTimeoutError as error:
+                        invocation.exception = error
+                        for result in invocation.results:
+                            result.result = ResultOutcome.ERROR
+
+                    except tmt.utils.RestartMaxAttemptsError as error:
+                        invocation.exception = error
```

Comment:
Seems, here we're doing the same exception handling three times. Perhaps we could catch all three exceptions as a list in a single `except` clause to deduplicate?

Reply:
Thanks!